### ISSUE TITLE
Kafka consumer always exit by several minutes

### ISSUE THREAD

**Issue by dashng** [segmentio/kafka-go#577]
**Describe the bug**
my code below, very sample, but always exited aftre running several minutes
```
package kafka

import (
	"context"
	"fmt"
	"log"

	"github.com/sgmentio/kafka-go"
)

func KafkaStart() {
	conf := kafka.ReaderConfig{
		Brokers:  []string{"xx.xxx.xx.xx:9092"},
		Topic:    "netflow_data",
		GroupID:  "g4",
		MaxBytes: 10,
	}

	reader := kafka.NewReader(conf)

	defer reader.Close()

	for {
		m, err := reader.ReadMessage(context.Background())
		if err != nil {
			log.Fatal(err)
		}
		fmt.Println("Message is" + string(m.Value))
	}
}

```
```
package main

import (
	"fmt"
	"time"

	appKafka "./kafka"
)

func main() {
	go appKafka.KafkaStart()
	fmt.Println("Kafka has been started ...")
	time.Sleep(10 * time.Minute)
}

```

**Kafka Version**
kafka_2.12-2.5.0

**Expected behavior**
loop consume data without exit

**Comment by Luffy110** [comment]
Did you always exit after ten minutes? if yes, i think it is due to you were using go routine to run KafkaStart() function, and the main thread sleep 10 minutes. so after 10 minutes sleep in main function, then your main thread will exit.

**Comment by CodyGuo** [comment]
close takes long time
```go
        r := kafka.NewReader(kafka.ReaderConfig{
                Brokers: address,
                GroupID: "logstash",
                Topic:   "tomcat-1",
        })
        now := time.Now()
        r.Close()
        fmt.Printf("closed: %s\n", time.Since(now))

```
output:
```log
closed: 3.140232992s
```
